### PR TITLE
GFA import

### PR DESCRIPTION
**Pull request by dkhofer**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -6,6 +6,7 @@ edition = "2021"
 [dependencies]
 bio = "2.0.0"
 clap = { version = "4.5.8", features = ["derive"] }
+gfa-reader = { git = "https://github.com/MoinSebi/gfa-reader", branch = "master" }
```
Comment (Chris7) [inline Cargo.toml:9]: can we do a commit hash?

**Review by Chris7 — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (dkhofer) [reply Cargo.toml]: Yep, just added that

File: src/imports/gfa.rs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+    let gfa: Gfa<u64, (), ()> = Gfa::parse_gfa_file(gfa_path);
+    let mut sequences_by_segment_id: HashMap<u64, Sequence> = HashMap::new();
+
+    for segment in &gfa.segments {
+        let input_sequence = segment.sequence.get_string(&gfa.sequence);
+        let sequence = Sequence::new()
+            .sequence_type("DNA")
+            .sequence(input_sequence)
+            .save(conn);
+        sequences_by_segment_id.insert(segment.id, sequence);
```
Comment (Chris7) [inline src/imports/gfa.rs:27]: we'll probably need to trim this out at larger scale as we won't be able to store all the sequences in memory

File: src/imports/gfa.rs [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+
+fn import_gfa(gfa_path: &str, collection_name: &str, conn: &Connection) {
+    models::Collection::create(conn, collection_name);
+    let block_group = BlockGroup::create(conn, collection_name, None, "");
+    let gfa: Gfa<u64, (), ()> = Gfa::parse_gfa_file(gfa_path);
+    let mut sequences_by_segment_id: HashMap<u64, Sequence> = HashMap::new();
+
+    for segment in &gfa.segments {
+        let input_sequence = segment.sequence.get_string(&gfa.sequence);
+        let sequence = Sequence::new()
```
Comment (Chris7) [inline src/imports/gfa.rs:23]: what would a shallow version of this look like?

File: src/imports/gfa.rs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+        path_edge_ids.push(edge_id);
+        Path::create(conn, path_name, block_group.id, path_edge_ids);
+    }
+
+    for input_walk in &gfa.walk {
+        let path_name = &input_walk.sample_id;
+        let mut source_hash = Sequence::PATH_START_HASH;
+        let mut source_coordinate = 0;
+        let mut path_edge_ids = vec![];
+        for segment_id in input_walk.walk_id.iter() {
```
Comment (Chris7) [inline src/imports/gfa.rs:116]: does this bring any of the strand info with it? walks do have direction

**Review by Chris7 — APPROVED** [review]
just some random comments on things i think we may have to do.

File: src/imports/gfa.rs (hunk 4 above)
Comment (dkhofer) [reply src/imports/gfa.rs:116]: I was being lazy.  I'll check what the parser provides and deal with it in a followup PR

**Pull request merged**